### PR TITLE
Added 3 new CTFs in problem-sets-competitive-programming.md

### PR DESCRIPTION
**Pull request by kn1-gh7**
## What does this PR do?
Add resource(s)

### Description
Adds 3 always on CTFs 
### Why is this valuable (or not)?
A lot of challenges to practice for CTF
### How do we know it's really free?
echoCTF and ROP wargame are available on github. The ringzer0ctf is on website and freely accessible after logging in.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: more/problem-sets-competitive-programming.md [hunk 1]
```diff
@@ -64,6 +65,8 @@
 * [Microcorruption](https://microcorruption.com/login)
 * [Overthewire Wargames fungame to practice CTF](https://overthewire.org/wargames/bandit)
 * [Picoctf](https://picoctf.org/resources.html) (email address *requested*)
+* [ringzer0ctf](https://ringzer0ctf.com/home) (email address *required*)
+* [ROP Wargame Repository](https://github.com/xelenonz/game)
```
Comment (eshellman) [inline more/problem-sets-competitive-programming.md:69]: Can you explain what this site is, and how it's a CTF site?

Comment (kn1-gh7) [reply more/problem-sets-competitive-programming.md:69]: A CTF (Capture the flag) is a security based competition consisting of challenges in various subdomains of security. The ringzer0CTF and echoCTF are CTF sites that keep on growing their challenge repository for people to practice for CTFs.
The ROP Wargame Repository is similiarly a github repository, consisting of challenges based on ROP(Return Oriented Programming) which is used in binary exploitation category of CTFs.

Comment (Ajay-singh1) [reply more/problem-sets-competitive-programming.md:69]: You are adding the links in Competitive Site Section.Maybe the link will find its place in another section kindly search through it.

Comment (kn1-gh7) [reply more/problem-sets-competitive-programming.md:69]: > You are adding the links in Competitive Site Section.Maybe the link will find its place in another section kindly search through it.

I added it in capture the flag subcategory. The name of the file indicates competitive programming but the file is no longer only that. As @eshellman pointed in PR [#10556](https://github.com/EbookFoundation/free-programming-books/pull/10556), changing the name of the file might cause conflicts.

**Pull request merged**